### PR TITLE
Auto chat migration (#37)

### PR DESCRIPTION
**Pull request by OlegKDev**

### CODE REVIEW

File: src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public ResultK<CommandProcessed, CommandError> process(Update update) {
+        final String oldToken = String.valueOf(update.message().migrateFromChatId());
+        final String newToken = String.valueOf(update.message().chat().id());
+
+        try {
+            roomService.autoMigrateToSupergroup(oldToken, newToken);
+        } catch (Exception e) {
+            return ResultK.Err(new CommandError(update.message().chat().id(), e.getMessage()));
```
Comment (Denis-26) [inline src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java:35]: Bad practice to send user exception message, better to create some readable massage like - something went wrong and log exception message with logger

File: src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        return ResultK.Ok(new CommandProcessed(update.message().chat().id(), "Chat was upgraded to supergroup"));
```
Comment (Denis-26) [inline src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java:38]: maybe more correct "Chat has been upgraded to supergroup"

Comment (OlegKDev) [reply src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java]: Done

File: src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java (hunk 1 above)
Comment (OlegKDev) [reply src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java]: Done

File: src/main/java/ru/all_easy/push/room/service/RoomService.java [hunk 3]
```diff
@@ -237,4 +238,22 @@ public void setRoomCurrency(RoomEntity room, CurrencyEntity currency) {
...
+        RoomEntity oldRoomEntity = repository.findByToken(oldToken);
+        RoomEntity newRoomEntity = new RoomEntity();
+        newRoomEntity.setTitle(oldRoomEntity.getTitle());
+        if (oldRoomEntity.getCurrency() != null) newRoomEntity.setCurrency(oldRoomEntity.getCurrency());
+        newRoomEntity.setToken(newToken);
+        repository.save(newRoomEntity);
+        autoMigrate(oldToken, newToken);
+    }
+
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
```
[the remark this answers is not in the document]
Comment (OlegKDev) [reply src/main/java/ru/all_easy/push/room/service/RoomService.java]: Done

**Pull request merged**